sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -431,16 +592,27 @@ class RocksDB(
   /**
    * Replay change log from the loaded version to the target version.
    */
-  private def replayChangelog(endVersion: Long): Unit = {
+  private def replayChangelog(
+      endVersion: Long,
+      stateStoreCkptIdLineage: Option[Array[LineageItem]] = None): Unit = {
+
+    val versionsAndUniqueIds = stateStoreCkptIdLineage match {
+      // First entry of lineage corresponds to loadedVersion
+      case Some(lineage) => lineage.map(i => (i.version, Some(i.checkpointUniqueId)))
+      case None => (loadedVersion + 1 to endVersion).map((_, None)).toArray
+    }
+
```

Comment:
I feel that the part generating `versionsAndUniqueIds` should be moved outside  `replayChangelog()`, and only `versionsAndUniqueIds` is passed in.

Reply:
sure change